### PR TITLE
Remove hardcoding of Ubuntu

### LINKED ISSUE #25: Do not hardcode Ubuntu
The recent pull request #21 hard coded ubuntu in the code.

Hard coding for a single Linux distribution is not the way to go and we eventually want to test on/for other distributions as well. Please remove it.

### PR DESCRIPTION
**Pull request by shkodm**
Resolves #25 
On Travis we still test only on it (16.04 and 18.04) , so I keep it as only choices for `system_testing.py` arguments.

### CODE REVIEW

File: push.py [hunk 1]
```diff
@@ -17,8 +17,8 @@
 parser.add_argument('-b', '--branch', help="log choosen preCICE branch")
 parser.add_argument('-t', '--test', help="choose system tests you want to use")
 parser.add_argument('-s', '--success', action='store_true' ,help="only upload log file")
-parser.add_argument('-o', '--os', type=str,help="Version of Ubuntu to use", choices =
-            ["1804", "1604"], default= "1604")
+parser.add_argument('-o', '--os', type=str,help="Distribution to use", choices =
```
Comment (floli) [inline push.py]: Is that really a distribution or the named variant of the base docker file? Do you adhere to https://github.com/precice/systemtests/issues/14?

**Review by floli — CHANGES_REQUESTED** [review]

File: push.py (hunk 1 above)
Comment (shkodm) [reply push.py]: I read #14, but here the same argument as for `system_testing.py` applies - right now we don't test anything else, but  'Ubuntu1604' and 'Ubuntu1804' on Travis, and I believe with @MakisH  we concluded that it is enough to test on different distributions. So `distributions` makes more sense to me.
 But if we decide to test and push results from different variants  ( as in #14 ) based on the same operating system, I would change this bit.

Comment (floli) [reply push.py]: Well, if we don't have the infrastructure in place we will never be able to test something else, therefore we don't do it.

We should definitely be able to test on arbitrary base images, as laid out in #14. If you feel that this scheme is not suitable, please comment here.

For travis only testing Ubuntu is ok, but the test suite should test other variants as well. None of supercomputers use Ubuntu.

Comment (MakisH) [reply push.py]: I confirm @shkodm that we agreed (between the two of us) that we need to be able on different distributions and that for Travis Ubuntu is enough. I am not sure I understand the problem here and what should be done differently.

Comment (shkodm) [reply push.py]: @floli, Allright, you are right, I will adjust that bit.

**Comment by floli** [comment]
What is the status of this PR? I have added an Arch based preCICE image and will eventually also try to add solver testing images for it. As far as I see, `push.py` is still Ubuntu only.

**Review by floli — APPROVED** [review]

**Pull request merged**